### PR TITLE
Add support dspi for s32z270dc2

### PR DESCRIPTION
**Pull request by tunguyen4585**
This PR to add support dspi for s32z27x devices
Tested on tests\drivers\spi\spi_loopback\drivers.spi.s32z_dspi.loopback
------ TESTSUITE SUMMARY START ------
SUITE PASS - 100.00% [spi_loopback]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.366 seconds
PASS - [spi_loopback.test_spi_loopback] duration = 0.366 seconds
------ TESTSUITE SUMMARY END ------

### CODE REVIEW

File: s32/mcux/devices/S32Z270/S32Z270_device.h
Comment (manuargue) [inline s32/mcux/devices/S32Z270/S32Z270_device.h:1956]: same comment as before, can we avoid reformatting the file after copied over from mcux-sdk?

**Review by manuargue — CHANGES_REQUESTED** [review]

File: s32/mcux/devices/S32Z270/fsl_device_registers.h
Comment (manuargue) [inline s32/mcux/devices/S32Z270/fsl_device_registers.h:19]: >add essential header files to build with mcux

elaborate pls, why this is needed?

File: dts/nxp/s32/S32Z27-BGA594-pinctrl.h
Comment (manuargue) [inline dts/nxp/s32/S32Z27-BGA594-pinctrl.h:615]: why there are no prefix for the port/pin on these three pinmuxes?

**Review by manuargue — CHANGES_REQUESTED** [review]

File: s32/mcux/devices/S32Z270/S32Z270_device.h
Comment (manuargue) [reply s32/mcux/devices/S32Z270/S32Z270_device.h:1956]: ignore this

File: s32/mcux/devices/S32Z270/fsl_device_registers.h
Comment (tunguyen4585) [reply s32/mcux/devices/S32Z270/fsl_device_registers.h:19]: + In _"S32Z2.h"_ included _"S32Z2_SPI.h"_ that has the struct **SPI_Type** and the macros as same as which I has defined in  _modules\hal\nxp\s32\mcux\devices\S32Z270\S32Z270_device.h_ and **mcux dspi** is using, leads to raise error re-defined. If we want to build DSPI with mcux, we cannot include "S32Z2_SPI.h"

+ If we include <S32Z2.h>, we could add unnecessary files to build

That why I want to include the essential files

I propose 2 solutions:
+ do like which implemented
+ jus remove or guard _#include "S32Z2_SPI.h"_ in _modules\hal\nxp\s32\drivers\s32ze\BaseNXP\header\S32Z2.h_ and keep them all

File: dts/nxp/s32/S32Z27-BGA594-pinctrl.h
Comment (tunguyen4585) [reply dts/nxp/s32/S32Z27-BGA594-pinctrl.h:615]: those pin belongs to LVDS pins and quite special, 
I see that we named the macros by "port" "_" "function"
however, I see in the IOMUX sheet the port is same as function :). Should we keep the same which i implemented?

File: s32/mcux/devices/S32Z270/S32Z270_device.h
Comment (Dat-NguyenDuy) [inline s32/mcux/devices/S32Z270/S32Z270_device.h:2121]: I thought RTD already defines all macros we need? isn't it

**Review by Dat-NguyenDuy — CHANGES_REQUESTED** [review]
As discussed with @tunguyen4585, we need to patch mcux-sdk DSPI driver, blocking this PR until the patch is merged and picked up into this PR

File: s32/mcux/devices/S32Z270/S32Z270_device.h
Comment (tunguyen4585) [reply s32/mcux/devices/S32Z270/S32Z270_device.h]: I remove all the RTD unsupported macros

**Comment by tunguyen4585** [comment]
> As discussed with @tunguyen4585, we need to patch mcux-sdk DSPI driver, blocking this PR until the patch is merged and picked up into this PR

The PR on mcux-sdk was merged, wait it sync to github

**Comment by Dat-NguyenDuy** [comment]
> As discussed with @tunguyen4585, we need to patch mcux-sdk DSPI driver, blocking this PR until the patch is merged and picked up into this PR

You'll need to update Patch List from [README](https://github.com/zephyrproject-rtos/hal_nxp/blob/master/mcux/README#L67) as well

**Comment by tunguyen4585** [comment]
> > As discussed with @tunguyen4585, we need to patch mcux-sdk DSPI driver, blocking this PR until the patch is merged and picked up into this PR
> 
> You'll need to update Patch List from [README](https://github.com/zephyrproject-rtos/hal_nxp/blob/master/mcux/README#L67) as well

I updated

**Review by Dat-NguyenDuy — APPROVED** [review]

**Review by manuargue — CHANGES_REQUESTED** [review]
block until zephyr pr approved

File: dts/nxp/s32/S32Z27-BGA594-pinctrl.h
Comment (manuargue) [reply dts/nxp/s32/S32Z27-BGA594-pinctrl.h:615]: I would suggest to use to use `LVDS_` as prefix for the voltage differential pads, to indicate that they are not targetting single-ended GPIO pads that belong to a port/pin pair

Comment (tunguyen4585) [reply dts/nxp/s32/S32Z27-BGA594-pinctrl.h]: I updated it, thank

File: s32/mcux/devices/S32Z270/fsl_device_registers.h
Comment (manuargue) [reply s32/mcux/devices/S32Z270/fsl_device_registers.h]: I see you have changed the approach in https://github.com/zephyrproject-rtos/hal_nxp/pull/435/commits/83f04dc0fdf09bc3e0b92bccee2810c039043cb1 
patches to the HAL must be documented in the s32/README

Comment (tunguyen4585) [reply s32/mcux/devices/S32Z270/fsl_device_registers.h]: I updated

Comment (manuargue) [reply s32/mcux/devices/S32Z270/fsl_device_registers.h]: I must say that I prefer the previous approach, including only what's necessary in zephyr's soc.h, but we can do it in a separate pr that address all drivers.

Comment (tunguyen4585) [reply s32/mcux/devices/S32Z270/fsl_device_registers.h]: yes, we will update in the future PRs.

**Review by manuargue — APPROVED** [review]

**Review by Dat-NguyenDuy — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Comment by manuargue** [comment]
@tunguyen4585 please rebase

**Review by manuargue — APPROVED** [review]

**Review by Dat-NguyenDuy — APPROVED** [review]

**Comment by manuargue** [comment]
zephyr pr approved by assignee, merging

**Pull request merged**
